### PR TITLE
th/dhcp-client

### PR DESCRIPTION
**Pull request by thom311**
Some fixes for ipv4.dhcp-client-id and ipv6.dhcp-duid

### CODE REVIEW

File: libnm-core/nm-utils.c [hunk 1]
```diff
@@ -4448,11 +4448,10 @@ _nm_utils_inet6_is_token (const struct in6_addr *in6addr)
 gboolean
```
Comment (fgiudici) [inline libnm-core/nm-utils.c]: Typos in the commit msg:
"Also, [is/it] is called from [...]"
[...]
"hug[e] settings, which could trick NetworkManage[r] to allocate [...]"

File: src/devices/nm-device.c [hunk 2]
```diff
@@ -7881,51 +7900,42 @@ dhcp6_get_duid (NMDevice *self, NMConnection *connection, GBytes *hwaddr, NMDhcp
...
 
-	} else if (nm_streq0 (duid, "stable-uuid")) {
-		duid_out = generate_duid_uuid (sha256_digest, len);
+		goto out_good;
 	}
 
-	duid_error = "generation failed";
-end:
-	if (!duid_out) {
+	g_return_val_if_reached (NULL);
```
Comment (fgiudici) [inline src/devices/nm-device.c]: We want to log the error but provide the backup random DUID here: wouldn't be better have just an nm_assert here?

**Comment by thom311** [comment]
All comments by Francesco addresses (via IRC)

**Comment by bengal** [comment]
LGTM

**Pull request merged**